src/scripts/underground/DailyDeal.ts

```diff
@@ -39,7 +39,27 @@ class DailyDeal {
     }
 
     private isValid(dealList: Array<DailyDeal>): boolean {
-        return ( (this.item1.name !== this.item2.name) && !DailyDeal.reverseDealExists(this.item1.name, this.item2.name, dealList) && !this.item1.isStone() );
+        const item1Name = this.item1.name;
+        const item2Name = this.item2.name;
+
+        if (item1Name !== item2Name) {
+            return false;
+        }
+
+        if (DailyDeal.sameDealExists(item1Name, item2Name, dealList)) {
+            return false;
+        }
+
+        return ( !DailyDeal.reverseDealExists(item1Name, item2Name, dealList) && !this.item1.isStone() );
```

Comment:
For the sake of consistency we should separate these 2 as you've done with the above 2 checks.